infra/README.md

```diff
@@ -0,0 +1,55 @@
+# DeltaNYC Infrastructure
+
+Infrastructure is maintained using;
+
+* [Terraform](https://www.terraform.io/)
+* [Cloud-Init](http://cloudinit.readthedocs.io/en/latest/index.html)
+
+
+# To the droplet's ip address
+
+```bash
+$ curl -s -X GET -H "Content-Type: application/json" -H "Authorization: Bearer $DIGITALOCEAN_TOKEN" "https://api.digitalocean.com/v2/droplets" | jq '.[] | .[] | select(.name=="delta_lookup_tool") | .networks.v4 | .[]'
+```
+
+We're currently using the DO ubuntu image, which allows ssh directly to the root account.
+
+```bash
+ssh -i ~/.ssh/id_rsa_delta root@{public_id}
+```
+
+The key to the machine is maintained by Chris Henry
+
+
+# Configuration
+
+### Google Maps API Key.
+
+The key in use was generated from Chris Henry's Google API Console, but any key
+with Google Maps permissions can be used.
+
+# Setup steps
+
+We should be using luigid, but I can't quite figure out what it's supposed to be
+doing.
```

Comment:
luigid is a server that handles scheduling and only scheduling. It doesn't know anything about possible tasks.

When a worker starts, it (not the server) pulls a task, computes the dependencies, and checks which are completed. It then submits all incomplete tasks to the scheduler, which then dictates to the worker which task to run.

So, luigi-import-worker is the image that does the actual import work and is the one that needs to be run; luigid is just a server that can orchestrate multiple workers running simultaneously. One potential gotcha is that docker-compose scale doesn't appear to start dependencies of the scaling service (?). I had to `docker-compose up luigid` prior to `docker-compose scale luigi-import-worker=x`. I'd guess the reason is that the scaling of one worker type shouldn't affect the scaling of the others and hence won't even check if the others have workers running at all.